doc/Programming-guide/Key-platform-features/Endpoint-groups/index.md

```diff
@@ -6,166 +6,308 @@ nav: /:path/Programming-guide/Key-platform-features/Endpoint-groups
+In case a group has no filter assigned for a specific profile schema version, the group does not apply to the endpoints that use the profile of this schema version.
 
 ## Profile filters ##
 
-
-Profile filters in Kaa are based on the [Spring Expression Language](http://docs.spring.io/spring/docs/3.0.x/reference/expressions.html) (SpEL). All filters must be specified as predicates (statements which may be either true or false). For further reference on the filters syntax, please refer to the Spring documentation.
+Profile filters in Kaa are based on the [Spring Expression Language](http://docs.spring.io/spring/docs/3.0.x/reference/expressions.html) (SpEL). 
+All filters must be specified as predicates (statements which may be either true or false). 
+For further reference on the filters syntax, please refer to the Spring documentation.
 Profile filters are evaluated using following context variables: 
 
-
 * "cp" - Client-side endpoint profile
 * "sp" - Server-side endpoint profile
 * "ekh" - Endpoint key hash
 
-## Profile filter examples ##
+### Profile filter examples ###
 
 The following example illustrates the general idea of profile filters.
 
+1. Let's assume the following client-side profile schema.
 
-* Let's assume the following client-side profile schema.
+```json
+[  
+   {  
+      "name":"ClientSideEndpointProfileChild",
+      "namespace":"org.kaaproject.kaa.common.endpoint.gen",
+      "type":"record",
+      "fields":[  
+         {  
+            "name":"otherSimpleField",
+            "type":"int"
+         },
+         {  
+            "name":"stringField",
+            "type":"string"
+         }
+      ]
+   },
+   {  
+      "namespace":"org.kaaproject.kaa.common.endpoint.gen",
+      "type":"record",
+      "name":"ClientSideEndpointProfile",
+      "fields":[  
+         {  
+            "name":"simpleField",
+            "type":"string"
+         },
+         {  
+            "name":"recordField",
+            "type":"org.kaaproject.kaa.common.endpoint.gen.ClientSideEndpointProfileChild"
+         },
+         {  
+            "name":"arraySimpleField",
+            "type":{  
+               "type":"array",
+               "items":"string"
+            }
+         },
+         {  
+            "name":"arrayRecordField",
+            "type":{  
+               "type":"array",
+               "items":"org.kaaproject.kaa.common.endpoint.gen.ClientSideEndpointProfileChild"
+            }
+         },
+         {  
+            "name":"nullableRecordField",
+            "type":[  
+               "org.kaaproject.kaa.common.endpoint.gen.ClientSideEndpointProfileChild",
+               "null"
+            ]
+         }
+      ]
+   }
+]
+```
 
+2. Let's assume the following server-side profile schema. Please note that this schema is less complex only for demonstration purposes. 
+Both Client-side and Server-side profile schemas support same level of complexity.
```

Comment:
"Both Client-side and Server-side profile schemas support same level of complexity."

Is that a feature by itself? The profiles should be flexible and support virtually any configuration. The complexity is a side-effect of this and should not be a feature. 

Can this be re-written?